### DISCUSSION TITLE
Cannot run test on lambda

### DISCUSSION THREAD

**Discussion by AppLightCode** [artilleryio/artillery#1723] [Artillery]
I am trying to run my test on lambda, but I am not able to run the test.

Here is my test yml and the error response:

**test.yml**

config:
  target: "--removed--"
  phases:
      - duration: 1
        arrivalRate: 1
        rampTo: 1
scenarios:
  - name: "Add Item"
    flow:
      - post:
          url: "/item"
          json:
            itemId: "123"
            product: "product test"
            price: "111"

**terminal output**

$ artillery run --platform aws:lambda --count 1 test.yml
NOTE: AWS Lambda support is experimental. Not all Artillery features work yet.
For details please see https://docs.art/aws-lambda

λ Creating AWS Lambda function...
    - Installing dependencies
C:\Users\Lenovo\AppData\Roaming\nvm\v18.12.1\node_modules\artillery\lib\platform\aws-lambda\index.js:104
      artillery.log(stdout.toString(), stderr.toString(), status, error);
                           ^

TypeError: Cannot read properties of null (reading 'toString')
    at PlatformLambda.init (C:\Users\Lenovo\AppData\Roaming\nvm\v18.12.1\node_modules\artillery\lib\platform\aws-lambda\index.js:104:28)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Launcher.run (C:\Users\Lenovo\AppData\Roaming\nvm\v18.12.1\node_modules\artillery\lib\launch-platform.js:351:25)

Node.js v18.12.1

**Comment by hassy** [comment]
What versiof of Artillery is this @AppLightCode? (`artillery version`) Could you try with the most recent version please?

**Reply by AppLightCode** [reply]
$ artillery version

        ___         __  _ ____
  _____/   |  _____/ /_(_) / /__  _______  __ ___ 
 /____/ /| | / ___/ __/ / / / _ \/ ___/ / / /____/
/____/ ___ |/ /  / /_/ / / /  __/ /  / /_/ /____/ 
    /_/  |_/_/   \__/_/_/_/\___/_/   \__  /       
                                    /____/        


VERSION INFO:

Artillery Core: 2.0.0-23
Artillery Pro:  not installed (https://artillery.io/product)

Node.js: v18.12.1
OS:      win32

**Reply by AppLightCode** [reply]
Will try with 27 now

**Comment by AppLightCode** [comment]
Lenovo@LAPTOP-N9OQOKFT MINGW64 /c/Lightcode/Load Testing/source/artillery.lambda
$ artillery version

        ___         __  _ ____
  _____/   |  _____/ /_(_) / /__  _______  __ ___
 /____/ /| | / ___/ __/ / / / _ \/ ___/ / / /____/
/____/ ___ |/ /  / /_/ / / /  __/ /  / /_/ /____/
    /_/  |_/_/   \__/_/_/_/\___/_/   \__  /
                                    /____/


VERSION INFO:

Artillery Core: 2.0.0-27
Artillery Pro:  not installed (https://artillery.io/product)

Node.js: v18.12.1
OS:      win32


Lenovo@LAPTOP-N9OQOKFT MINGW64 /c/Lightcode/Load Testing/source/artillery.lambda
$ artillery run --platform aws:lambda --count 1 test.yml
NOTE: AWS Lambda support is experimental. Not all Artillery features work yet.
For details please see https://docs.art/aws-lambda

λ Creating AWS Lambda function...
- Bundling test data
  - test.yml
  - package.json
  - package-lock.json
- Installing dependencies
undefined undefined null Error: spawnSync npm ENOENT
    at Object.spawnSync (node:internal/child_process:1110:20)
    at spawnSync (node:child_process:857:24)
    at PlatformLambda.init (C:\Users\Lenovo\AppData\Roaming\nvm\v18.12.1\node_modules\artillery\lib\platform\aws-lambda\index.js:147:47)
    at async Launcher.run (C:\Users\Lenovo\AppData\Roaming\nvm\v18.12.1\node_modules\artillery\lib\launch-platform.js:358:25) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawnSync npm',
  path: 'npm',
  spawnargs: [ 'install' ]
}
- Creating zip package
Preparing AWS environment...
 - Lambda function: artilleryio-07b7f5a1-3b38-4002-8bb3-a26b3ec3aa79
 - Region: us-east-1
 - AWS account: #######
[bf4c9c4e-cc4b-441a-b2e4-db15356a7b13]: A Lambda function has exited with an error. Reason: ArtilleryError
{
  stdout: '',
  stderr: 'node:internal/modules/cjs/loader:936\n' +
    '  throw err;\n' +
    '  ^\n' +
    '\n' +
    "Error: Cannot find module '@oclif/command'\n" +
    'Require stack:\n' +
    '- /var/task/node_modules/artillery/bin/run\n' +
    '    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)\n' +
    '    at Function.Module._load (node:internal/modules/cjs/loader:778:27)\n' +
    '    at Module.require (node:internal/modules/cjs/loader:1005:19)\n' +
    '    at require (node:internal/modules/cjs/helpers:102:18)\n' +
    '    at Object.<anonymous> (/var/task/node_modules/artillery/bin/run:7:17)\n' +
    '    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n' +
    '    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n' +
    '    at Module.load (node:internal/modules/cjs/loader:981:32)\n' +
    '    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n' +
    '    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12) {\n' +
    "  code: 'MODULE_NOT_FOUND',\n" +
    "  requireStack: [ '/var/task/node_modules/artillery/bin/run' ]\n" +
    '}\n'
}
All VUs finished. Total time: 5 days, 12 hours, 56 minutes, 37 seconds

--------------------------------
Summary report @ 14:56:37(+0200)
--------------------------------

No measurements recorded during this period

Estimated AWS Lambda cost for this test: $0.0031

**Reply by hassy** [reply]
It looks like `npm` is either not installed or is not in $PATH - could you confirm by running "npm" in the same terminal window as the one you're using to run Artillery?

**Comment by AppLightCode** [comment]
Lenovo@LAPTOP-N9OQOKFT MINGW64 /c/Lightcode/Load Testing/source/artillery.lambda
$ node -v
v18.12.1

Lenovo@LAPTOP-N9OQOKFT MINGW64 /c/Lightcode/Load Testing/source/artillery.lambda
$ npm -v
8.19.2

Lenovo@LAPTOP-N9OQOKFT MINGW64 /c/Lightcode/Load Testing/source/artillery.lambda
$ artillery run --platform aws:lambda --count 1 test.yml
NOTE: AWS Lambda support is experimental. Not all Artillery features work yet.
For details please see https://docs.art/aws-lambda

λ Creating AWS Lambda function...
- Bundling test data 
  - test.yml
  - package.json     
  - package-lock.json
- Installing dependencies
undefined undefined null Error: spawnSync npm ENOENT
    at Object.spawnSync (node:internal/child_process:1110:20)
    at spawnSync (node:child_process:857:24)
    at PlatformLambda.init (C:\Users\Lenovo\AppData\Roaming\nvm\v18.12.1\node_modules\artillery\lib\platform\aws-lambda\index.js:147:47)
    at async Launcher.run (C:\Users\Lenovo\AppData\Roaming\nvm\v18.12.1\node_modules\artillery\lib\launch-platform.js:358:25) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawnSync npm',
  path: 'npm',
  spawnargs: [ 'install' ]
}
- Creating zip package
Preparing AWS environment...
 - Lambda function: artilleryio-616d33a1-2c8f-472d-8e67-6b2931ebb4fc
 - Region: us-east-1
 - AWS account: #######
[354d6d5c-0afa-46a3-a421-0ca9729c18c4]: A Lambda function has exited with an error. Reason: ArtilleryError
{
  stdout: '',
  stderr: 'node:internal/modules/cjs/loader:936\n' +
    '  throw err;\n' +
    '  ^\n' +
    '\n' +
    "Error: Cannot find module '@oclif/command'\n" +
    'Require stack:\n' +
    '- /var/task/node_modules/artillery/bin/run\n' +
    '    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)\n' +
    '    at Function.Module._load (node:internal/modules/cjs/loader:778:27)\n' +
    '    at Module.require (node:internal/modules/cjs/loader:1005:19)\n' +
    '    at require (node:internal/modules/cjs/helpers:102:18)\n' +
    '    at Object.<anonymous> (/var/task/node_modules/artillery/bin/run:7:17)\n' +
    '    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n' +
    '    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n' +
    '    at Module.load (node:internal/modules/cjs/loader:981:32)\n' +
    '    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n' +
    '    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12) {\n' +
    "  code: 'MODULE_NOT_FOUND',\n" +
    "  requireStack: [ '/var/task/node_modules/artillery/bin/run' ]\n" +
    '}\n'
}
All VUs finished. Total time: 6 days, 14 hours, 11 minutes, 51 seconds

--------------------------------
Summary report @ 16:11:51(+0200)
--------------------------------

No measurements recorded during this period

Estimated AWS Lambda cost for this test: $0.0002

**Comment by kavinarvind** [comment]
Facing the same error when trying to run artillery over lambda. Checked the npm path and its newer artillery version. Is it some open issue @hassy
